File: agents/api_server.py
#!/usr/bin/env python3
" Flask API Server for Kanban Agent Integration
import asyncio
from flask import Flask, request, jsonify
from flask_cors import CORS
import threading, sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).parent))
from kanban_review_fix_agent import KanbanReviewFixAgent, AgentConfig
app = Flask(__name__)
CORS(app)
kanban_agent = None
agent_lock = threading.Lock()
def get_kanban_agent():
    global kanban_agent
    with agent_lock:
        if kanban_agent is None:
            config = AgentConfig(name=KanbanReviewFixAgent, description=Fix stuck cards, system_prompt=Fix stuck cards, tools=[scan,fix,report])
 kanban_agent = KanbanReviewFixAgent(config)
 return kanban_agent
@app.route(/api/v1/health)
def health(): return jsonify({status:healthy})
@app.route(/api/v1/kanban/scan, methods=[POST])
def scan():
 try:
 agent = get_kanban_agent()
 loop = asyncio.new_event_loop()
 asyncio.set_event_loop(loop)
 async def run():
 await agent.initialize()
 r = await agent.execute_task({action:scan})
 await agent.shutdown()
 return r
 r = loop.run_until_complete(run())
 loop.close()
 return jsonify({success:r.success,data:r.data,errors:r.errors or []})
 except Exception as e: return jsonify({success:False,error:str(e)}),500
@app.route(/api/v1/kanban/fix, methods=[POST])
def fix():
 try:
 agent = get_kanban_agent()
 data = request.get_json() or {}
 loop = asyncio.new_event_loop()
 asyncio.set_event_loop(loop)
 async def run():
 await agent.initialize()
 task = {action:fix}
 if data.get(card_id): task[card_id]=data[card_id]
 r = await agent.execute_task(task)
 await agent.shutdown()
 return r
 r = loop.run_until_complete(run())
 loop.close()
 return jsonify({success:r.success,data:r.data,errors:r.errors or []})
 except Exception as e: return jsonify({success:False,error:str(e)}),500
@app.route(/api/v1/kanban/report, methods=[POST])
def report():
 try:
 agent = get_kanban_agent()
 loop = asyncio.new_event_loop()
 asyncio.set_event_loop(loop)
 async def run():
 await agent.initialize()
 r = await agent.execute_task({action:report})
 await agent.shutdown()
 return r
 r = loop.run_until_complete(run())
 loop.close()
 return jsonify({success:r.success,data:r.data,errors:r.errors or []})
 except Exception as e: return jsonify({success:False,error:str(e)}),500
if __name__==__main__:
 print(Starting Kanban Agent API Server on http://localhost:5000)
 app.run(host=0.0.0.0,port=5000,debug=False)

File: agents/pkg/api_server.go
package pkg

import (
	"encoding/json"
	"net/http"
	"sync"

	"github.com/gorilla/mux"
	"github.com/rs/cors"
)

// APIServer provides HTTP API for agent integration
type APIServer struct {
	kanbanAgent *KanbanReviewFixAgent
	agentLock   sync.Mutex
	config      *AgentConfig
}

// NewAPIServer creates a new APIServer
func NewAPIServer(config *AgentConfig) *APIServer {
	return &APIServer{
		config: config,
	}
}

// getKanbanAgent returns singleton kanban agent instance
func (s *APIServer) getKanbanAgent() *KanbanReviewFixAgent {
	s.agentLock.Lock()
	defer s.agentLock.Unlock()

	if s.kanbanAgent == nil {
		s.kanbanAgent = NewKanbanReviewFixAgent(s.config, "config/kanban-config.json")
	}

	return s.kanbanAgent
}

// HealthHandler returns server health status
func (s *APIServer) HealthHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]string{
		"status": "healthy",
	})
}

// ScanHandler runs kanban scan operation
func (s *APIServer) ScanHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	agent := s.getKanbanAgent()

	if err := agent.Initialize(); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	result, err := agent.ExecuteTask(map[string]interface{}{
		"action": "scan",
	})

	agent.Shutdown()

	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		json.NewEncoder(w).Encode(map[string]interface{}{
			"success": false,
			"error":   err.Error(),
		})
		return
	}

	json.NewEncoder(w).Encode(map[string]interface{}{
		"success": result.Success,
		"data":    result.Data,
		"errors":  result.Errors,
	})
}

// FixHandler runs kanban fix operation
func (s *APIServer) FixHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	var data map[string]interface{}
	if err := json.NewDecoder(r.Body).Decode(&data); err != nil {
		data = make(map[string]interface{})
	}

	agent := s.getKanbanAgent()

	if err := agent.Initialize(); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	task := map[string]interface{}{
		"action": "fix",
	}

	if cardID, ok := data["card_id"].(string); ok && cardID != "" {
		task["card_id"] = cardID
	}

	result, err := agent.ExecuteTask(task)

	agent.Shutdown()

	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		json.NewEncoder(w).Encode(map[string]interface{}{
			"success": false,
			"error":   err.Error(),
		})
		return
	}

	json.NewEncoder(w).Encode(map[string]interface{}{
		"success": result.Success,
		"data":    result.Data,
		"errors":  result.Errors,
	})
}

// ReportHandler generates kanban report
func (s *APIServer) ReportHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")

	agent := s.getKanbanAgent()

	if err := agent.Initialize(); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	result, err := agent.ExecuteTask(map[string]interface{}{
		"action": "report",
	})

	agent.Shutdown()

	if err != nil {
		w.WriteHeader(http.StatusInternalServerError)
		json.NewEncoder(w).Encode(map[string]interface{}{
			"success": false,
			"error":   err.Error(),
		})
		return
	}

	json.NewEncoder(w).Encode(map[string]interface{}{
		"success": result.Success,
		"data":    result.Data,
		"errors":  result.Errors,
	})
}

// Start starts the API server on specified address
func (s *APIServer) Start(addr string) error {
	r := mux.NewRouter()

	// API routes
	r.HandleFunc("/api/v1/health", s.HealthHandler).Methods("GET")
	r.HandleFunc("/api/v1/kanban/scan", s.ScanHandler).Methods("POST")
	r.HandleFunc("/api/v1/kanban/fix", s.FixHandler).Methods("POST")
	r.HandleFunc("/api/v1/kanban/report", s.ReportHandler).Methods("POST")

	// CORS configuration
	c := cors.New(cors.Options{
		AllowedOrigins:   []string{"*"},
		AllowedMethods:   []string{"GET", "POST", "OPTIONS"},
		AllowedHeaders:   []string{"Content-Type"},
		AllowCredentials: true,
	})

	handler := c.Handler(r)

	return http.ListenAndServe(addr, handler)
}